src/utils.rs

```diff
@@ -0,0 +1,60 @@
+#[cfg(any(target_os = "linux", target_os = "freebsd"))]
+use crate::WireguardInterfaceError;
+#[cfg(any(target_os = "linux", target_os = "freebsd"))]
+use std::{
+    io::Write,
+    process::{Command, Stdio},
+};
+
+#[cfg(any(target_os = "linux", target_os = "freebsd"))]
+pub(crate) fn set_dns(ifname: &str, dns: Vec<String>) -> Result<(), WireguardInterfaceError> {
+    // Build the resolvconf command
+    let mut cmd = Command::new("resolvconf");
+    let args = ["-a", ifname, "-m", "0", "-x"];
+    debug!("Executing comamnd resolvconf with args: {:?}", args);
+    cmd.args(args);
+
+    let search_entries: Vec<String> = dns
```

Comment:
I'm sure this whole parsing process can be shortened, but maybe it's not necessary if we switch to using actual IPs.